storio-common/src/main/java/com/pushtorefresh/storio/TypeMappingFinder.java

```diff
@@ -0,0 +1,93 @@
+package com.pushtorefresh.storio;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Common class for search and cache indirect type mapping.
+ */
+public class TypeMappingFinder {
```

Comment:
Let's extract interface from this class and rename implementation to `impl`? Then put interface to public `com.pushtorefresh.storio` class and implementation to `com.pushtorefresh.storio.internal`